pylint/checkers/typecheck.py

```diff
@@ -1241,13 +1245,28 @@ def visit_compare(self, node):
         if op in ['in', 'not in']:
             self._check_membership_test(right)
 
-    @check_messages('unsubscriptable-object', 'unsupported-assignment-operation',
-                    'unsupported-delete-operation')
+    @check_messages('unsubscriptable-object',
+                    'unsupported-assignment-operation',
+                    'unsupported-delete-operation',
+                    'unhashable-dict-key')
     def visit_subscript(self, node):
         supported_protocol = None
         if isinstance(node.value, (astroid.ListComp, astroid.DictComp)):
             return
 
+        if isinstance(node.value, astroid.Dict):
```

Comment:
We should check the context as well, we only want this for Store, so all the time we try to stash a nonhashable variable in a dictionary